api/info/info.go

```diff
@@ -50,8 +51,22 @@ func (h *Handler) Handle(w http.ResponseWriter, r *http.Request) error {
 	if err != nil {
 		return err
 	}
+
+	var profile *config.SigningProfile
+
+	policy := h.sign.Policy()
+	if policy != nil && policy.Profiles != nil && req.Profile != "" {
+		profile = policy.Profiles[req.Profile]
+	}
+
+	if profile == nil && policy != nil {
+		profile = policy.Default
+	}
+
 	resp := client.InfoResp{
-		Certificate: bundler.PemBlockToString(&pem.Block{Type: "CERTIFICATE", Bytes: cert.Raw}),
+		Certificate:  bundler.PemBlockToString(&pem.Block{Type: "CERTIFICATE", Bytes: cert.Raw}),
+		Usage:        profile.Usage,
```

Comment:
I see, so I should be checking if profile is nil, rather than if policy is
nil?

On Wed, May 6, 2015 at 2:51 PM, Kyle Isom <notifications@github.com> wrote:

> In api/info/info.go
> <https://github.com/cloudflare/cfssl/pull/192#discussion_r29789464>:
>
> >  	resp := client.InfoResp{
> > -		Certificate: bundler.PemBlockToString(&pem.Block{Type: "CERTIFICATE", Bytes: cert.Raw}),
> > +		Certificate:  bundler.PemBlockToString(&pem.Block{Type: "CERTIFICATE", Bytes: cert.Raw}),
> > +		Usage:        profile.Usage,
>
> With profile, it *shouldn't* be nil after the initial setting (though it
> *could* be); a failure check should be done there. Then you can use the
> profile without worrying it it's nil.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cloudflare/cfssl/pull/192/files#r29789464>.
>



-- 
Alice Xia
Cornell University '16